### PR TITLE
refactor: move structs to args mod

### LINKED ISSUE #1243: Move NetworkOpts, RpcServerOpts to standalone `args` module
### Describe the feature

atm some helper `Args` are in `lib.rs`

## TODO

* move https://github.com/paradigmxyz/reth/blob/9470943babf8afe98a10db729954a374470afcb7/bin/reth/src/lib.rs#L30 to `src/args/network_args.rs`
* move https://github.com/paradigmxyz/reth/blob/9470943babf8afe98a10db729954a374470afcb7/bin/reth/src/lib.rs#L64 to `src/args/rpc_server_args.rs`
* rename types `Opts -> Args` to follow clap's naming
* make all fields `pub`

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by chirag-bgh**
fixes #1243

### CODE REVIEW

File: bin/reth/src/args/rpc_server_args.rs [hunk 1]
```diff
@@ -24,83 +24,49 @@ pub use reth_staged_sync::utils;
 use clap::Args;
 use reth_primitives::NodeRecord;
 
-/// Parameters for configuring the network more granularity via CLI
```
Comment (mattsse) [inline bin/reth/src/args/rpc_server_args.rs:27]: we need to keep lib and only move the types to a new module `args`

**Review by mattsse — CHANGES_REQUESTED** [review]

File: bin/reth/src/args/mod.rs [hunk 2]
```diff
@@ -0,0 +1,3 @@
+pub mod network_args;
+
+pub mod rpc_server_args;
```
Comment (mattsse) [inline bin/reth/src/args/mod.rs:3]: nit: use `mod` and reexport via `pub use network_args::NetworkArgs;`

**Review by mattsse — APPROVED** [review]
lgtm

pending @onbjerg

**Review by gakonst — APPROVED** [review]
good w me

**Pull request merged**